indico/modules/events/contributions/forms.py

```diff
@@ -130,3 +132,25 @@ def validate_duration(self, field):
             return
         if self.contrib.is_scheduled and self.contrib.start_dt + field.data > self.contrib.event_new.end_dt:
             raise ValidationError(_('With the current value, the contribution would exceed event end date'))
+
+
+class ContributionTypeForm(IndicoForm):
+    """Form to create or edit a ContributionType"""
+
+    name = StringField(_("Name"), [DataRequired()])
+    description = TextAreaField(_("Description"))
+
+    def __init__(self, *args, **kwargs):
+        self.event = kwargs.pop('event')
+        self.contrib_type = kwargs.pop('contrib_type', None)
+        if self.contrib_type:
```

Comment:
you can just pass `obj=self.contrib_type` (don't even need to wrap it in `FormDefaults` since we don't need additional fancy behavior) to get the correct defaults